lightning-invoice/src/lib.rs

```diff
@@ -1187,6 +1188,14 @@ impl Invoice {
 			.unwrap_or(Duration::from_secs(DEFAULT_EXPIRY_TIME))
 	}
 
+	/// Returns whether the invoice has expired.
+	pub fn is_expired(&self) -> bool {
+		match self.timestamp().elapsed() {
+			Ok(elapsed) => elapsed > self.expiry_time(),
+			Err(_) => false,
```

Comment:
Should we call invoice with a timestamp in the future expired? I mean its clear we shouldn't be sending to it one way or another, I guess.

Reply:
I think this could possibly be caused by inconsistency in time between machines, IIUC. We wouldn't want to fail if off by a few seconds.

Reply:
Yea, true, I guess my mind always goes to "overflow", but that's a good point.